### DISCUSSION TITLE
Non-Maximum Suppression in flax/jax

### DISCUSSION THREAD

**Discussion by wztdream** [google/flax#1929] [Q&A] [answered]
Hi,
 I noticed in comment https://github.com/google/flax/pull/427#issuecomment-744435259 says that:

> The only place I would expect errors to show up from is the post-processing work (NMS and Top-K selection), which was not possible to implement in Jax

Is it true that we can not using flax/jax to implement NMS? It seems to me we can directly use jax.numpy to implement it, am I wrong? Is there special concern if we use jax.numpy to implement NMS?

Is there official model implemented by flax that using NMS for reference?

**Comment by marcvanzee** [comment]
Hi @wztdream, I don't know the context of that comment (it is very old), but I am sure we are able to implement this in JAX somehow. Perhaps @Marvin182 knows more about this, since he was involved in that project?

I don't know any reference implementation though, but if you have any specific questions feel free to ask.

**Comment by levskaya** [comment, marked as answer]
@wztdream - there's actually an NMS implementation in the older JAX MLPerf Training 0.7 "SSD" entry at:
https://github.com/mlperf/training_results_v0.7/blob/master/Google/benchmarks/ssd/implementations/ssd-research-JAX-tpu-v3-4096/nms.py

Note that some the of other neural net code there uses the deprecated pre-linen Flax API, but the NMS implementation is pure JAX, so hopefully will help.

**Reply by wztdream** [reply]
this helps a lot, thank you
